source/simulator/freesurface.cc

```diff
@@ -172,17 +210,35 @@ namespace aspect
       VectorTools::interpolate_boundary_values (free_surface_dof_handler, *p,
                                                 ZeroFunction<dim>(dim), mesh_displacement_constraints);
 
-    //Zero out the displacement for the prescribed velocity
+    // Zero out the displacement for the prescribed velocity boundaries
+    // if the boundary is not in the set of tangential mesh boundaries
     for (std::map<types::boundary_id, std::pair<std::string, std::string> >::const_iterator p = sim.parameters.prescribed_velocity_boundary_indicators.begin();
          p != sim.parameters.prescribed_velocity_boundary_indicators.end(); ++p)
-      VectorTools::interpolate_boundary_values (free_surface_dof_handler, p->first,
-                                                ZeroFunction<dim>(dim), mesh_displacement_constraints);
+      {
+        if (tangential_mesh_boundary_indicators.find(p->first) == tangential_mesh_boundary_indicators.end())
+          {
+            VectorTools::interpolate_boundary_values (free_surface_dof_handler, p->first,
+                                                      ZeroFunction<dim>(dim), mesh_displacement_constraints);
+          }
+      }
+
+    // Zero out the displacement for the traction boundaries
+    // if the boundary is not in the set of tangential mesh boundaries
+    for (std::map<types::boundary_id, std::pair<std::string, std::string> >::const_iterator p = sim.parameters.prescribed_traction_boundary_indicators.begin();
```

Comment:
This logic is new, right? That is to say, we were not previously handling traction boundaries correctly here?

I think that your choice of defaulting to zero displacement is a good one.